src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -301,30 +378,73 @@ public int getIdleMinutes() {
         return (idleMinutes != null) ? idleMinutes : 0;
     }
 
+    @DataBoundSetter
+    public void setIdleMinutes(Integer idleMinutes) {
+        this.idleMinutes = (idleMinutes != null) ? idleMinutes : 0;
+    }
+
     public int getMaxSize() {
-        return maxSize;
+        return Math.max(1, maxSize);
+    }
+
+    @DataBoundSetter
+    public void setMaxSize(int maxSize) {
+        if (maxSize < minSize) {
+            int newMaxSize = minSize > 0 ? minSize : 1;
+            warning("Cloud parameter 'maxSize' can't be less than 'minSize' or 1, setting to %d", newMaxSize);
+            maxSize = newMaxSize;
+        }
+        this.maxSize = Math.max(1, maxSize);
```

Comment:
`Math.max(minSize, maxSize)` ?

Reply:
Using `Math.max(minSize, maxSize)` could cause problems if `minSize` is 0.